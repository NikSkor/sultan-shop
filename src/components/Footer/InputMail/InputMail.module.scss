@import '../../../vars.scss';

.form {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 13px 10px 13px 20px;
  gap: 10px;
  background-color: #FFFFFF;
  border-radius: 36px;
  width: 282px;
  height: 59px;

  @include laptop {
    grid-area: form;
  }
}

.searchInput {
  border: none;
  background-color: #FFFFFF;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  color: #5C6370;
  opacity: 0.6;
}

.btn {
  cursor: pointer;
}