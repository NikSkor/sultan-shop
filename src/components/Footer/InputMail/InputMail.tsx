import React, { FC } from 'react';
import style from './InputMail.module.scss';

const InputMail: FC = () => {
  return (
    <form className={style.form}>
      <input className={style.searchInput} type="text" placeholder='Введите ваш E-mail' />
      <button className={style.btn}>
        <svg width="39" height="39" viewBox="0 0 39 39" fill="none" xmlns="http://www.w3.org/2000/svg">
          <rect width="39" height="39" rx="19.5" fill="#FFC85E"/>
          <path d="M16 24.8571L21 19.5L16 14.1429L17 12L24 19.5L17 27L16 24.8571Z" fill="white"/>
        </svg>
      </button>
    </form>
  )
}

export default InputMail;