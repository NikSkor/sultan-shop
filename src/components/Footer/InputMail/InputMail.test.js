import { render, screen } from '@testing-library/react';
import InputMail from './InputMail';

describe('test of inputMail component', () => {
  test('telephone', () => {
    render(<InputMail />);
    const mail = screen.getByPlaceholderText(/Введите ваш E-mail/i);
    expect(mail).toBeInTheDocument();
    expect(mail).toMatchSnapshot();
  });

  test('button mail', () => {
    render(<InputMail />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });
});
