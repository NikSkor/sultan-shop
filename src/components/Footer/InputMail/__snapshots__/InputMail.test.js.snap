// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of inputMail component button mail 1`] = `
<button
  class="btn"
>
  <svg
    fill="none"
    height="39"
    viewBox="0 0 39 39"
    width="39"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="#FFC85E"
      height="39"
      rx="19.5"
      width="39"
    />
    <path
      d="M16 24.8571L21 19.5L16 14.1429L17 12L24 19.5L17 27L16 24.8571Z"
      fill="white"
    />
  </svg>
</button>
`;

exports[`test of inputMail component telephone 1`] = `
<input
  class="searchInput"
  placeholder="Введите ваш E-mail"
  type="text"
/>
`;
